types/rke_types.go

```diff
@@ -110,8 +110,8 @@ type PrivateRegistry struct {
 	Password string `yaml:"password" json:"password,omitempty" norman:"type=password"`
 	// Default registry
 	IsDefault bool `yaml:"is_default" json:"isDefault,omitempty"`
-	// CredentialPlugin
-	CredentialPlugin map[string]string `yaml:"credentialPlugin" json:"credentialPlugin,omitempty"`
+	// ECRCredentialPlugin
+	ECRCredentialPlugin *ECRCredentialPlugin `yaml:"ecrCredentialPlugin" json:"ecrCredentialPlugin,omitempty"`
```

Comment:
To be consistent in notation, shouldn't yaml tag be is snake_case??

```
-	ECRCredentialPlugin *ECRCredentialPlugin `yaml:"ecrCredentialPlugin" json:"ecrCredentialPlugin,omitempty"`
+	ECRCredentialPlugin *ECRCredentialPlugin `yaml:"ecr_credential_plugin" json:"ecrCredentialPlugin,omitempty"`
```

Reply:
It can be either way. I could not find a predefined preference after looking at the other types. Majority of the types had yaml annotations in snake_case while json ones are camel_case.

Reply:
sorry missed the point earlier.. fixed it up